src/config.rs

```
@@ -49,13 +227,220 @@ fn load_raw_config(filepath: String) -> RawConfig {
         }
     };
 
-    let r: RawConfig = toml_value;
+    let mut r: RawConfig = toml_value;
+
+    // Expand variables in the fields
+    expand_raw_config_fields(&mut r, env_option);
+
     r
 }
 
+fn expand_raw_config_fields(r: &mut RawConfig, e: &Option<HashMap<String,String>>) {
+    expand_raw_option_string(&mut r.edf_system_search_path, e);
+    expand_raw_option_string(&mut r.parallax_imagestore, e);
+    expand_raw_option_string(&mut r.parallax_mount_program, e);
+    expand_raw_option_string(&mut r.parallax_path, e);
+    expand_raw_option_string(&mut r.podman_module, e);
+    expand_raw_option_string(&mut r.podman_path, e);
+    expand_raw_option_string(&mut r.podman_tmp_path, e);
+    expand_raw_option_string(&mut r.runtime_path, e);
+    expand_raw_option_string(&mut r.tracking_tool, e);
+}
+
+fn expand_raw_option_string(optstr: &mut Option<String>, env_option: &Option<HashMap<String,String>>) {
+    if optstr.is_some() {
+        *optstr =
+            Some(expand_vars_string(optstr.clone().unwrap(), env_option).unwrap_or(String::from("")));
+    }
+}
```

Should we raise an error if some environment variables are not bound (e.g., because of typos)? System configs are critical to functionality, so I think some pedantic sanity checks might be valuable as a precaution.

I think this can be done by adding `set -u` in the command (`-c`) of `expand_vars_string_with_env` and making `expand_raw_option_string` to return `SarusResult`.

---

I agree, but how to do that ?
Just returning an error, we are invalidating the configuration load for something that maybe is not really important.
We cannot just print a warning, because we are in a library and we have no way to handle I/O.
We can return an error with the wrong expanded value inside together with a warning message, and the caller can decide what to do with it.

What do you think ?

---

I think the last option sounds reasonable. I was just wondering what would be the best way to deliver the error message to the caller. Do you have any idea?